web-admin/src/features/projects/github/GithubRepoSelectionDialog.svelte

```diff
@@ -0,0 +1,197 @@
+    DialogDescription,
+    DialogFooter,
+    DialogHeader,
+    DialogTitle,
+    DialogTrigger,
+  } from "@rilldata/web-common/components/dialog-v2";
+  import {
+    Collapsible,
+    CollapsibleTrigger,
+    CollapsibleContent,
+  } from "@rilldata/web-common/components/collapsible";
+  import { Button } from "@rilldata/web-common/components/button";
+  import Input from "@rilldata/web-common/components/forms/Input.svelte";
+  import Github from "@rilldata/web-common/components/icons/Github.svelte";
+  import Select from "@rilldata/web-common/components/forms/Select.svelte";
+  import Spinner from "@rilldata/web-common/features/entity-management/Spinner.svelte";
+  import { EntityStatus } from "@rilldata/web-common/features/entity-management/types";
+  import { eventBus } from "@rilldata/web-common/lib/event-bus/event-bus";
+  import CaretDownIcon from "@rilldata/web-common/components/icons/CaretDownIcon.svelte";
+  import CaretUpIcon from "@rilldata/web-common/components/icons/CaretUpIcon.svelte";
+  import type { AxiosError } from "axios";
+
+  export let open = false;
+  export let currentUrl: string;
+  export let currentSubpath: string;
+  export let currentBranch: string;
+  export let project: string;
+  export let organization: string;
+
+  let advancedOpened = false;
+
+  const githubData = getGithubData();
+  const userRepos = githubData.userRepos;
+  const status = githubData.status;
+  const projectQuery = createAdminServiceGetProject(organization, project);
+
+  $: repoSelections =
+    $userRepos.data?.repos?.map((r) => ({
+      value: r.url,
+      label: `${r.owner}/${r.name}`,
+    })) ?? [];
+
+  const githubConnectionUpdater = new GithubConnectionUpdater();
+  // update data from project, this is needed if the user never leaves the status page and this component is not unmounted
+  $: githubConnectionUpdater.init(currentUrl, currentSubpath, currentBranch);
+
+  const connectToGithubMutation =
+    githubConnectionUpdater.connectToGithubMutation;
+  const showOverwriteConfirmation =
+    githubConnectionUpdater.showOverwriteConfirmation;
+  let githubUrl = githubConnectionUpdater.githubUrl;
+  let subpath = githubConnectionUpdater.subpath;
+  let branch = githubConnectionUpdater.branch;
+
+  function onRepoChange(newUrl: string) {
+    const repo = $userRepos.data?.repos?.find((r) => r.url === newUrl);
+    if (!repo) return; // shouldnt happen
+
+    githubConnectionUpdater.onRepoChange(repo);
+  }
+  $: if (!$githubUrl && repoSelections.length === 1) {
+    onRepoChange(repoSelections[0].value);
+  }
+
+  $: if ($showOverwriteConfirmation) {
+    // hide the selection dialog if overwrite confirmation dialog is open
+    open = false;
+  }
+
+  async function updateGithubUrl(force: boolean) {
+    if (
+      !(await githubConnectionUpdater.update({
+        organization,
+        project,
+        force,
+        instanceId: $projectQuery.data?.prodDeployment?.runtimeInstanceId ?? "",
+      }))
+    ) {
```

Comment:
Nit: Embedding the mutation into the conditional check makes this hard to read. I'd recommend declaring a self-describing boolean, like `didUpdate`.